grokability/snipe-it
# Allow CompanyableTrait trait on users

This should allow us to automatically scope users to company instead of having to manually do it. Previously, adding the `use CompanyableTrait;` statement would get caught in an infinite loop, since `Auth::check()` internally calls `Auth::user()`, which makes the scope get caught on *which* user you want to be looking at for scoping. 

Using `Auth::hasUser()` just returns the boolean value, which prevents this looping problem.

### To test

- enable full multiple company support
- create a few users in a few different companies. give them permission to see assets, users, etc, but not superadmin
- create some assets in those same companies
- login as each of those users, and see that the results are scoped to the company the respective users belong to
